source/Stargate-Examples-Tests/PetStoreAPITest.class.st

```diff
@@ -47,28 +47,176 @@ PetStoreAPITest >> testBadRequest [
 ]
 
 { #category : #'tests - CORS' }
-PetStoreAPITest >> testCORSHeadersWhenOriginIsAllowed [
+PetStoreAPITest >> testCORSAllowingAnyOrigin [
 
 	| response |
 
-	api beCORSAwareAllowing: {self baseUrl}.
+	api allowCrossOriginSharingApplying: [ :cors | cors allowAnyOrigin ].
+		
+	response := self newClient
+		url: self baseUrl / 'pets' asUrl;
+		headerAt: 'Origin' put: self baseUrl printString;
+		headerAt: 'Access-Control-Request-Headers' put: 'Content-Type';
+		options;
+		response.
+		
+	self 	
+		assert: response code equals: 204;
+		assert: ( response headers at: 'Access-Control-Allow-Origin' ) equals: '*';
+		assert: ( response headers at: 'Access-Control-Allow-Headers' ) equals: 'Content-Type';
+		assert: ( response headers at: 'Vary' ) equals: 'Access-Control-Allow-Headers';
+		assert: ( response headers at: 'Access-Control-Allow-Methods' ) equals:'POST, GET'.
```

Comment:
`response accessControlAllowMethods` can return a method collection instead of the formatted version :). We're doing something like this for `Link` headers.

Reply:
you mean as an extension?